examples/vision/mlpmixer_image_classification.py

```diff
@@ -0,0 +1,275 @@
+import tensorflow_addons as tfa
+
+"""
+## Prepare the data
+"""
+
+num_classes = 100
+input_shape = (32, 32, 3)
+
+(x_train, y_train), (x_test, y_test) = keras.datasets.cifar100.load_data()
+
+print(f"x_train shape: {x_train.shape} - y_train shape: {y_train.shape}")
+print(f"x_test shape: {x_test.shape} - y_test shape: {y_test.shape}")
+
+"""
+## Configure the hyperparameters
+"""
+
+learning_rate = 0.005
+weight_decay = 0.0001
+batch_size = 128
+num_epochs = 50
+dropout_rate = 0.2
+image_size = 64  # We'll resize input images to this size.
+patch_size = 8  # Size of the patches to be extracted from the input images.
+num_patches = (image_size // patch_size) ** 2  # Size of the data array.
+hidden_units = 256  # Number of hidden units.
+num_mixers = 4  # Number of mixer blocks.
+
+print(f"Image size: {image_size} X {image_size} = {image_size ** 2}")
+print(f"Patch size: {patch_size} X {patch_size} = {patch_size ** 2} ")
+print(f"Patches per image: {num_patches}")
+print(f"Elements per patch (3 channels): {(patch_size ** 2) * 3}")
+
+"""
+## Use data augmentation
+"""
+
+data_augmentation = keras.Sequential(
+    [
+        layers.experimental.preprocessing.Normalization(),
+        layers.experimental.preprocessing.Resizing(image_size, image_size),
+        layers.experimental.preprocessing.RandomFlip("horizontal"),
+        layers.experimental.preprocessing.RandomZoom(
+            height_factor=0.2, width_factor=0.2
+        ),
+    ],
+    name="data_augmentation",
+)
+# Compute the mean and the variance of the training data for normalization.
+data_augmentation.layers[0].adapt(x_train)
+
+"""
+## Implement the Multi-layer Perceptron (MLP) module
+"""
+
+
+def create_mlp(hidden_units, dropout_rate):
+    mlp = keras.Sequential(
+        [
+            layers.Dense(units=hidden_units, activation=keras.activations.gelu),
+            layers.Dense(units=hidden_units),
+            layers.Dropout(rate=dropout_rate),
+        ]
+    )
+    return mlp
+
+
+"""
+## Implement patch creation as a layer
+"""
+
+
+class Patches(layers.Layer):
+    def __init__(self, patch_size, num_patches):
+        super(Patches, self).__init__()
+        self.patch_size = patch_size
+        self.num_patches = num_patches
+
+    def call(self, images):
+        batch_size = tf.shape(images)[0]
+        patches = tf.image.extract_patches(
+            images=images,
+            sizes=[1, self.patch_size, self.patch_size, 1],
+            strides=[1, self.patch_size, self.patch_size, 1],
+            rates=[1, 1, 1, 1],
+            padding="VALID",
+        )
+        patch_dims = patches.shape[-1]
+        patches = tf.reshape(patches, [batch_size, self.num_patches, patch_dims])
+        return patches
+
+
+"""
+## Implement the patch encoding layer
+
+The `PatchEncoder` layer will linearly transform a patch by projecting it into
+a vector of size `hidden_units`. Note that no positional encoding is used.
+"""
+
+
+class PatchEncoder(layers.Layer):
```

Comment:
This is just a `Dense` layer. You can remove this layer entirely and just use `Dense`.

Reply:
+1